### ISSUE TITLE
Validator errors on slicing

### ISSUE THREAD

**Issue by ahenket** [FirelyTeam/fhir-net-api#640]
When you validate this file at https://simplifier.net/validate, you will get these errors. It is as if the fixed values for the slices aren't recognized. The ICPC error is also incorrect.

![image](https://user-images.githubusercontent.com/5390573/42877510-355c72ca-8a8a-11e8-8357-31c1d33fa526.png)

[medmij-gpdata-fhir3-0-1-observation-contact05-E.txt](https://github.com/ewoutkramer/fhir-net-api/files/2205502/medmij-gpdata-fhir3-0-1-observation-contact05-E.txt)

**Comment by ewoutkramer** [comment]
Yes, the slicing validation need work. This will be the focus of the next release (after 1.0) this fall.

**Comment by marcovisserFurore** [comment]
I've tested this validation with the new version of the fhir-net-api, version 1.4.0.

I got now the following validation information:
```
Overall result: FAILURE

[FATAL] Internal logic failure: The value discriminator should have either a 'fixed[x]' or 'binding' element set on 'http://hl7.org/fhir/StructureDefinition/Coding#Coding.code'. (at Observation)
```

@ahenket: can you verify if that is correct?

I've tested it with the following steps:
- Go to https://simplifier.net/nictizstu3-zib2017/$projectserver
- Download the server to local machine
- Unpack the Fhir Server and start it with the command `start-vonk-server.ps1`
- POST the example resource in this issue to http://localhost:8080/Observation/$validate

**Comment by ewoutkramer** [comment]
This is actually the grandfather of #1185 and #1246 - this has been fixed and will be released in 1.7.
